sqlglot/transforms.py

```diff
@@ -12,14 +12,17 @@ def unalias_group(expression):
     """
     if isinstance(expression, exp.Group) and isinstance(expression.parent, exp.Select):
         aliased_selects = {
-            e.alias: i for i, e in enumerate(expression.parent.expressions, start=1) if isinstance(e, exp.Alias)
+            e.alias: (i, e.this)
+            for i, e in enumerate(expression.parent.expressions, start=1)
+            if isinstance(e, exp.Alias)
         }
 
         expression = expression.copy()
 
         for col in expression.find_all(exp.Column):
-            alias_index = aliased_selects.get(col.name)
-            if not col.table and alias_index:
+            alias_index, col_expression = aliased_selects.get(col.name, (None, None))
+            group_by_expression = col.find_ancestor_with_parent(exp.Group)
```

Comment:
Would it be simpler to just iterate over the immediate children of the Group expression?

i.e.
```python
for col in expression.children:
    if isinstance(col, exp.Column) and ...:
        ...
```

Then we don't need to traverse down and back up the tree.

Reply:
Nice suggestion! I made that change in the latest commit.